at_lookup/lib/src/at_lookup_impl.dart

```diff
@@ -319,25 +317,31 @@ class AtLookupImpl implements AtLookUp {
     } on Exception catch (e) {
       logger.severe('Error in remote verb execution ${e.toString()}');
       var errorCode = AtLookUpExceptionUtil.getErrorCode(e);
-      return Future.error(AtLookUpException(
-          errorCode, AtLookUpExceptionUtil.getErrorDescription(errorCode)));
+      throw AtLookUpException(
+          errorCode, AtLookUpExceptionUtil.getErrorDescription(errorCode));
     }
+    // If connection time-out, do not return empty verbResult;
+    // throw AtLookupException.
+    if (verbResult.isEmpty) {
+      throw AtLookUpException('AT0014', 'Request timed out');
+    }
+    // If response starts with error:, throw AtLookupException.
     if (_isError(verbResult)) {
       verbResult = verbResult.replaceAll('error:', '');
-      var errorCode = verbResult.split('-')[0];
-      var errorMessage = verbResult.split('-')[1];
-      return Future.error(AtLookUpException(errorCode, errorMessage));
+      throw AtLookUpException(
+          verbResult.split('-')[0], verbResult.split('-')[1]);
```

Comment:
Please add defensive code for when the verbResult doesn't contain '-' just in case in future we have a badly behaved server which returns an error code without an error description

Reply:
Added a defensive code when server does not return error response with '-'